crates/iota-network/src/discovery/mod.rs

```diff
@@ -74,7 +74,8 @@ pub struct NodeInfo {
 #[derive(Clone, Debug, Default)]
 /// Contains a new list of available trusted peers.
 pub struct TrustedPeerChangeEvent {
-    pub new_peers: Vec<PeerInfo>,
+    pub new_committee: Vec<PeerInfo>,
+    pub old_committee: Vec<PeerInfo>,
```

Comment:
Any reason not to use HashSets here?

Reply:
We don't need it here, the type returned by `get_validator_as_p2p_peers` is Vec, PeerInfo is neither Eq, nor Hash.